springwolf-core/src/main/java/io/github/stavshamir/springwolf/schemas/example/DefaultSchemaWalker.java

```diff
@@ -202,25 +191,34 @@ private T handleObject(String name, Schema schema, Map<String, Schema> definitio
         return exampleValueGenerator.createEmptyObjectExample();
     }
 
-    private T handleObjectProperties(
-            String name, Map<String, Schema> properties, Map<String, Schema> definitions, Set<Schema> visited) {
-
-        List<PropertyExample<T>> propertyList = properties.entrySet().stream()
+    private List<PropertyExample<T>> buildPropertyListFromSchema(
+            Map<String, Schema> properties, Map<String, Schema> definitions, Set<Schema> visited) {
+        return properties.entrySet().stream()
+                // TODO: comparing to sort? Remove or add to buildPropertyListFromSchemas
                 .sorted(Map.Entry.comparingByKey())
                 .map(entry -> {
                     String propertyKey = entry.getKey();
                     T propertyValue = buildSchemaInternal(propertyKey, entry.getValue(), definitions, visited);
                     return new PropertyExample<>(propertyKey, propertyValue);
                 })
                 .toList();
+    }
 
-        return exampleValueGenerator.createObjectExample(name, propertyList);
+    private List<PropertyExample<T>> buildPropertyListFromSchemas(
```

Comment:
I would vote for a different method name. The current naming is in line with buildSchemaInternal, but I also find this misleading. So maybe that should also include renaming the buildSchemaInternal...